### PR TITLE
VC-4249: added methods for MsgSystemRewardTransfer

### PR DESCRIPTION
**Pull request by at-kh**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by vasylNaumenko — COMMENTED** [review]
-= LGTM =-

File: modules/vipcoin/chain/banking/msg_system_reward_transfer.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	transfers, err := m.bankingRepo.GetSystemTransfers(filter.NewFilter().SetArgument(dbtypes.FieldCreator, msg.Creator))
+	if err != nil {
+		return err
+	}
+
+	if len(transfers) != 1 {
+		return types.ErrInvalidCreatorField
+	}
+
+	time, err := strconv.ParseInt(tx.Timestamp, 10, 64)
```
Comment (alex-bodnar) [inline modules/vipcoin/chain/banking/msg_system_reward_transfer.go]: time, err := time.Parse(time.RFC3339, tx.Timestamp)
	if err != nil {
		return err
	}

File: modules/vipcoin/chain/banking/msg_system_reward_transfer.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+)
+
+// handleMsgSystemRewardTransfer allows to properly handle a MsgSystemRewardTransfer
+func (m *Module) handleMsgSystemRewardTransfer(tx *juno.Tx, index int, msg *types.MsgSystemRewardTransfer) error {
+	if err := m.bankingRepo.SaveMsgSystemRewardTransfers(msg); err != nil {
+		return err
+	}
+
+	transfers, err := m.bankingRepo.GetSystemTransfers(filter.NewFilter().SetArgument(dbtypes.FieldCreator, msg.Creator))
+	if err != nil {
```
Comment (alex-bodnar) [inline modules/vipcoin/chain/banking/msg_system_reward_transfer.go]: можно заменить на:

	switch {
	case err != nil:
		return err
	case len(transfers) != 1:
		return types.ErrInvalidCreatorField
	}

**Review by alex-bodnar — APPROVED** [review]
LGTM

**Pull request merged**